src/main.rs

```diff
@@ -1,3 +1,76 @@
+mod memory;
+mod opdcodes;
+mod registers;
+
+use memory::Memory;
+use opdcodes::Opcode;
+use registers::Registers;
+
+struct VM {
+    memory: Memory,
+    registers: Registers,
+}
+
+impl VM {
+    pub fn new() -> Self {
+        Self {
+            memory: Memory::new(),
+            registers: Registers::new(),
+        }
+    }
+
+    pub fn read_memory(&self, address: u16) -> u16 {
+        self.memory.read(address)
+    }
+
+    pub fn write_memory(&mut self, address: u16, value: u16) {
+        self.memory.write(address, value);
+    }
+
+    pub fn read_register(&self, r: usize) -> u16 {
+        self.registers.get(r).unwrap_or(0)
+    }
```

Comment:
Quick question, is returning a 0 the best option here? I'm not sure I understand the reason behind it.

I think having this function return a Result would aid you when the time for error handling comes. What do you think?

Reply:
Returning 0 is a horrible decision tbh,  I should have chosen to use a Result.

Reply:
Returning 0 is an horrible desicion tbh, i should had choosen using a Result